app/src/main/java/cloudcity/MainActivityExtensions.java

```diff
@@ -18,21 +18,46 @@ public class MainActivityExtensions {
 
     /**
      * We will be using the 'logging' shared pref since that's the only one that is displayed in the logging settings fragment...
-     * @param TAG
-     * @param spg
+     *
+     * @param TAG the log tag used for logging
+     * @param spg the {@link SharedPreferencesGrouper} on which we'll set the listener and from which we'll grab the logging shared pref
+     * @param dp the {@link DataProvider} which will be used to refresh all 'data' necessary for the CC server updates
      */
     public static void performMainActivityThing(String TAG, SharedPreferencesGrouper spg, DataProvider dp) {
         // First, set a listener that will trigger when the actual CC logging changes
         spg.setListener((sharedPreferences, key) -> {
+            if (key != null && !key.isEmpty() && CloudCityUtil.isBlank(key)) {
```

Comment:
oh sh... will fix.

Reply:
oh, no, we don't want to check non-empty strings whether they're blank. on pre-34 devices that will first trim the string then check if that's empty; which is unnecessary if the first isEmpty() fails. 

what i mean by "will fix" was the lack of "not-blank" since i noticed it just said `&& isBlank(string)`.

anyways. fixed. resolving.